src/test/java/io/xstefank/wildfly/bot/PRConfigFileChangeTest.java

```diff
@@ -0,0 +1,139 @@
+package io.xstefank.wildfly.bot;
+
+import io.quarkiverse.githubapp.testing.GitHubAppMockito;
+import io.quarkiverse.githubapp.testing.GitHubAppTest;
+import io.quarkus.test.junit.QuarkusTest;
+import io.xstefank.wildfly.bot.config.RuntimeConstants;
+import io.xstefank.wildfly.bot.model.MockedGHPullRequestFileDetail;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+import org.kohsuke.github.GHCommitState;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHEvent;
+import org.kohsuke.github.GHPullRequestFileDetail;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.PagedSearchIterable;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+
+import static io.quarkiverse.githubapp.testing.GitHubAppTesting.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+@GitHubAppTest
+public class PRConfigFileChangeTest {
+
+    @Test
+    void testUpdateToIncorrectFile() throws IOException {
+        given().github(mocks -> {
+            GHRepository repo = mocks.repository("The-Huginn/wildfly-github-bot");
+            GHContent mockGHContent = mock(GHContent.class);
+            when(repo.getFileContent(".github/" + RuntimeConstants.CONFIG_FILE_NAME, "d4cb8c40c8d1d568bd12126c6eb1f541e06fee4b")).thenReturn(mockGHContent);
+            when(mockGHContent.read()).thenReturn(IOUtils.toInputStream( "wildfly:\n" +
+                    "  rules:\n" +
+                    "    - title: \"test\"\n" +
+                    "    - body: \"test\"\n" +
+                    "      notify: [The-non-existing-user]\n" +
+                    "  format:\n" +
+                    "    title-check:\n" +
+                    "      pattern: \"\\\\[TITLE-\\\\d+\\\\]\\\\s+.*|TITLE-\\\\d+\\\\s+.*\"\n" +
+                    "      message: \"Wrong content of the title!\"\n" +
+                    "    description:\n" +
+                    "      regexes:\n" +
+                    "        - pattern: \"\\\\s+https://\"\n" +
+                    "          message: \"The PR description must contain a link.\"\n" +
+                    "    commits-quantity:\n" +
+                    "      quantity: \"1-3\"\n" +
+                    "      message: \"Too many commits in PR!\"\n" +
+                    "  emails:\n" +
+                    "    - rastislavbudinsky@gmail.com\n" +
+                    "    - rbudinsk@redhat.com",
```

Comment:
you probably don't want real emails in tests

Reply:
Thanks for noticing.